### PR TITLE
Merge branch extreme precpitation tool

### PR DESCRIPTION
**Pull request by mukhtyar**
This branch was created by cherry picking commits related to the extreme precipitation tool. Additional commits include updating the tool to use the new helper functions for setting initial config and location.

This branch has been deployed to the dev site for review - https://dev.cal-adapt.org/tools/extreme-precipitation

### CODE REVIEW

File: src/routes/tools/extreme-precipitation/_ExploreData.svelte [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  let bookmark;
+  let learnMoreProps = {};
+  let metadata;
+  let csvData;
+  let printContainer;
+  let printSkipElements;
+
+  $: chartDescription = $indicator.description;
+
+  $: chartTitle = $location.title;
```
Comment (clhenrick) [inline src/routes/tools/extreme-precipitation/_ExploreData.svelte]: @mukhtyar I noticed the location isn't updating in the title, you'll probably need to use `svelte.afterUpdate`

File: src/routes/tools/extreme-precipitation/index.svelte [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+</div>
+
+<div class="bx--grid">
+  <div id="about" use:inview="{{}}" on:enter="{handleEntry}">
+    <About
+      datasets="{datasets}"
+      on:datasetLoaded="{(e) => datasetStore.set(e.detail)}"
+    >
+      <div slot="description">
+        <p>
```
Comment (clhenrick) [inline src/routes/tools/extreme-precipitation/index.svelte]: @mukhtyar would you mind moving the about description to `content/tools/extreme-precipitation/about.md` and adding it similarly to the other tools? You don't need to format it to Markdown, you can just copy & paste the HTML.

File: src/routes/tools/extreme-precipitation/index.svelte [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+          or 99th percentiles. The tool then displays the extreme events that
+          exceed the threshold in different ways. The Frequency chart shows the
+          estimated intensity of precipitation events (Return Level) for a
+          selected period (Return Period) and how it changes over the historical
+          period (1961–1990), mid-century (2035–2064) and end-century
+          (2071–2099). The other charts display the total number of events, the
+          timing of these events and the longest stretch of consecutive extreme
+          events.
+        </p>
+        <p class="h4">What is a Threshold value?</p>
```
Comment (clhenrick) [inline src/routes/tools/extreme-precipitation/index.svelte]: You could use `<h3>`s for these sub-headings as "About the Tool" is rendered as an `<h2>`.

File: src/routes/tools/extreme-precipitation/index.svelte [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+          climate zones in California. On a local scale different statistical
+          assumptions (i.e. using annual maximal values rather than partial
+          duration time-series, fitting techniques for distribution parameters
+          and choice of extreme value distribution) may be more appropriate. We
+          encourage users to ensure the empirical fit of the applied
+          distribution is acceptable to their end use before using estimates
+          produced from this tool for planning purposes.
+        </p>
+
+        <p class="h3">References</p>
```
Comment (clhenrick) [inline src/routes/tools/extreme-precipitation/index.svelte]: same here, this could also be an `<h3>`

**Review by clhenrick — COMMENTED** [review]
@mukhtyar I noticed that the location name in the title isn't updating. 

Also you could move the static content for About to a markdown file and preload it similarly to how the other tools are doing so now.

File: src/routes/tools/extreme-precipitation/index.svelte (hunk 2 above)
Comment (mukhtyar) [reply src/routes/tools/extreme-precipitation/index.svelte]: Fixed in https://github.com/berkeley-gif/caladapt-website-2021/pull/186/commits/4663c64c03b58e1420dbd5f5154152ab169122c2

File: src/routes/tools/extreme-precipitation/index.svelte (hunk 3 above)
Comment (mukhtyar) [reply src/routes/tools/extreme-precipitation/index.svelte]: Fixed in https://github.com/berkeley-gif/caladapt-website-2021/pull/186/commits/4663c64c03b58e1420dbd5f5154152ab169122c2

**Comment by mukhtyar** [comment]
Closing this as additional work and QA round are needed. Will create a new PR to merge final version of the Extreme Precipitation tool into the main branch.

**Pull request closed without merging**